.github/workflows/try-runtime.yml

```diff
@@ -0,0 +1,64 @@
+name: Try-runtime
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, edited]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  TRY_RUNTIME_CLI_VERSION_TAG: v0.7.0
+  CARGO_HOME: ./.cargo
+
+jobs:
+  check_skip_flags:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Commit
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ env.GITHUB_SHA }}
+      - name: Get Head Commit Message
+        id: head-commit-message
+        run: echo "HEAD_COMMIT_MESSAGE=$(git show -s --format=%s)" >> "$GITHUB_OUTPUT"
+
+  test-try-runtime:
+    strategy:
+      matrix:
+        runtime: [peregrine, spiritnet]
+    runs-on: ubuntu-latest
+    needs: check_skip_flags
+    container:
+      image: paritytech/ci-unified:bullseye-1.74.0
+
+    if: ${{ !contains(needs.check_skip_flags.outputs.head-commit-message, 'ci-skip-rust') }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up caching for Cargo
+        uses: actions/cache@v4
+        with:
+          path: ${{ secrets.CARGO_CACHE_PATH }}
+          key: cargo-try-runtime-${{ matrix.runtime }}-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
Why are `key` and `restore-keys` different?

Reply:
They are not different it's best practice to use just a prefix on `restore-key` since "When a key doesn't match directly, the action searches for keys prefixed with the restore key". Evenif there is change on the hash, use exsisting previous cache and update it for the next round. On the otherhand  if we try exact it will not find a cache at least for one round.